lib/variable/to_unit.cpp

```diff
@@ -47,8 +48,15 @@ Variable to_unit(const Variable &var, const units::Unit &unit,
         "This limitation exists because such conversions would require "
         "information about calendars and time zones.");
   }
-  return variable::transform(var, scale * unit, core::element::to_unit,
-                             "to_unit");
+  Variable scalevar;
+  if (const auto iscale = std::round(scale);
+      (var.dtype() == dtype<int64_t> ||
+       var.dtype() == dtype<core::time_point>)&&(std::abs(scale - iscale) <
+                                                 1e-12 * std::abs(scale)))
```

Comment:
Where does `1e-12` come from?

Reply:
It is somewhat arbitrary. Do you have a better suggestion?

Reply:
I was wondering if you did tests with a range of values to check if this tolerance works but a smaller one doesn't.

Reply:
What about
```.py
const auto src_scale = var_unit.underlying().<scale>();
const auto dst_scale = unit.underlying().<scale>();
if (<is-int-dtype> && fmod(src_scale, dst_scale) < DBL_EPSILON) {
    scalevar = static_cast<int64_t>(src_scale / dst_scale);
} else {...}
```
From https://stackoverflow.com/a/49220844/12010475

I honestly don't fully understand why this compares using `< DBL_EPSILON`. How can you get a number smaller than epsilon??
But this would avoid checking the quotient and operate on the scales directly which is more precise.

Reply:
> I honestly don't fully understand
Maybe we should avoid such a solution then.

Can you see any scenario where we have a unit scale ratio that is closer than 1e-12 to an integer but should *not* be treated as such?